### PR TITLE
Added Costa-Rica (#617)

### PR DESCRIPTION
**Pull request by mlucchini**
A few comments to make on this one:
- There is (close to) up-to-date production data for Costa-Rica that is available on one page but it doesn't give sources by type (wind, hydro, etc) but instead the names of the power stations and I didn't find a mapping between those. Also, to get up-to-date data you have to play with cookies and sessions as they use JSF which should be doable but is somewhat messy.
- There is the page I used which gives the production per source but it is for the whole of yesterday.
- No interchange yet.

### CODE REVIEW

**Comment by corradio** [comment]
Hi @mlucchini,
Thanks for the pull. The Electricity Map only considers data that is less than 2 hrs old, so fetching the day before won't be of much help.
I however believe we can crowdsource the mapping of power stations to fuel type.
If you make a list, I believe we could post that to Slack and get help from the community?

**Comment by mlucchini** [comment]
Updated the PR with real-time production data and 24h history as well as exchange with Panama. There is also an alternative CR->NI exchange now in CR.py that could be used instead of the CR->NI exchange from NI.py; not sure which one is the most reliable or up-to-date.

File: parsers/CR.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            'coal': 0.0,
+            'gas': 0.0,
+            'hydro': 0.0,
+            'nuclear': 0.0,
+            'oil': 0.0,
+            'solar': 0.0,
+            'wind': 0.0,
+            'geothermal': 0.0,
+            'unknown': 0.0,
+            None: 0.0
```
Comment (corradio) [inline parsers/CR.py]: You probably don't need that?

**Comment by corradio** [comment]
This sounds pretty cool. Can you update the README?

**Comment by corradio** [comment]
Also, I've got a suspicious measurement when I run the parser:
- Seems like the None key has data (should it be unknown?)
- Biomass is negative (shouldn't be the case!)

```python
{'oil': 0.0, 'gas': 0.0, 'biomass': -0.77000000000000002, 'geothermal': 456.13, 'solar': 0.0, 'unknown': 79.550000000000011, None: 3643.0799999999999, 'coal': 0.0, 'nuclear': 0.0, 'wind': 40.460000000000008, 'hydro': 3066.96}
```

**Comment by mlucchini** [comment]
None is essentially /dev/null, just a way to discard the Total present on each row.
Will look at the biomass issue.

**Comment by corradio** [comment]
mmh ok - can you discard the None key then? It will create some problems as this is not a valid production key.

**Comment by mlucchini** [comment]
Updated README, removed interchange rows classified as "unknown" and cleaned-up the None hack.
However the "El Viejo" biomass row appears negative on the [source of the data](https://appcenter.grupoice.com/CenceWeb/CencePosdespachoNacional.jsf), perhaps @jarek or @systemcatch can figure out why this is the case?

**Comment by mlucchini** [comment]
Or is El Viejo a hydro source absorbing energy surplus? Aeroenergía is also negative sometimes. Could replace negatives with 0s if that feels right to you guys.

**Comment by corradio** [comment]
Mmh we should look deeper into this. Negative production should be negated
and put into the storage dict if indeed it represents storing energy
surplus.

**Comment by systemcatch** [comment]
@mlucchini El Viejo is is a biomass plant located next to a big sugar refinery.  It burns the waste material from sugar production.  I don't think it has any storage capability.  My guess is that when the powerplant is idle it draws a small amount from the grid to keep some function going.  So we can probably safely ignore any negative values.

**Comment by mlucchini** [comment]
Thanks for this @systemcatch ! I will set the negatives to 0. There are a couple other ones which are sometimes negative such as Aeroenergía but the amounts are minuscule (0.0x) so I'll 0 those ones too.

**Comment by corradio** [comment]
Looks good to me!

**Pull request merged**